### ISSUE TITLE
Copy as SQL INSERT not working correctly for utf8mb4?

### ISSUE THREAD

**Issue by opinxit** [sequelpro/sequelpro#2348]
If I Copy as SQL INSERT on a row in a utf8mb4 table, I get a lot of garbage? for string values like the following (integers come through fine): 

...
VALUES
	(X'636F6C6C656374696F6E73', X'73706F6E735F6D72715F64617465', X'53706F6E736F72736869702045787069726174696F6E2044617465', X'', X'74657874', 0, 0, X'', X'', 0, 89, X'', 50, X'636C6173733D226461746574696D657069636B657222', 0, X'', X'', 0);

**Comment by dmoagx** [comment]
What exactly is wrong with the data?
```
mysql> select X'636F6C6C656374696F6E73', X'73706F6E735F6D72715F64617465', X'53706F6E736F72736869702045787069726174696F6E2044617465', X'', X'74657874', 0, 0, X'', X'', 0, 89, X'', 50, X'636C6173733D226461746574696D657069636B657222', 0, X'', X'', 0\G
*************************** 1. row ***************************
                                X'636F6C6C656374696F6E73': collections
                          X'73706F6E735F6D72715F64617465': spons_mrq_date
X'53706F6E736F72736869702045787069726174696F6E2044617465': Sponsorship Expiration Date
                                                      X'': 
                                              X'74657874': text
                                                        0: 0
                                                        0: 0
                                                      X'': 
                                                      X'': 
                                                        0: 0
                                                       89: 89
                                                      X'': 
                                                       50: 50
          X'636C6173733D226461746574696D657069636B657222': class="datetimepicker"
                                                        0: 0
                                                      X'': 
                                                      X'': 
                                                        0: 0
1 row in set (0,00 sec)

```

**Comment by opinxit** [comment]
Perhaps I should ask how you convert it?

**Comment by dmoagx** [comment]
Nothing. The data is simply HEX-encoded.
Sequel Pro usually does that if the field type is BLOB or uses binary collation, to preserve non-printable characters.
How does your column definition look like?

**Comment by opinxit** [comment]
Yes, indeed, it uses binary collation -- it just threw me that copy as sql insert would return values different from just plain copy.

**Comment by captain-digitalsailors** [comment]
My take on this: The "Copy as SQL INSERT" feature should not HEX-encode data if the field uses a binary collation. If a database field uses the utf-8 charset and one wants to have case sensitive string comparisons by default, the collation has to be utf-8-bin. As a lot of utf-8 characters can be displayed correctly by the system, having inserts with a lot of hex data is a nuisance, especially if I copied as SQL INSERT especially to edit that data.

**Comment by dmoagx** [comment]
Yes, a `bin` collation does not mean binary data.
Sequel Pro has an issue where it misinterprets what binary means. See #2700

**Comment by suheti** [comment]
Is there a option to try exporting it as human readable string, instead of hex coded? MySQL Workbench exports INSERT statements just as displayed. That's much easier to work with in most of my working scenario.

**Comment by neerajsinghsonu** [comment]
Hello @dmoagx ,

I had same issue while copying as 'Copy as SQL INSERT', My column was set as 'VARCHAR', still I was getting X'<Random Stuffs>. Then I just noticed that while moving on 'Strcture' tab for table, required column was ticked as Binary (I didnt did that). So just unchecked the Binary column and everything stared working again fine.

OS: MacOS High Sierra (10.13.6)
Sequal PRO: 1.1.2

Issue in action:
![2019-08-25_11-17-53](https://user-images.githubusercontent.com/1378886/63646106-7fe86000-c72a-11e9-8dca-1268b957e141.gif)




Regards,
Neeraj Singh
